DZ60 keymap and layout

OlliGranlund opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

<!--- Describe your changes in detail here. -->

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

drashna commented on .gitignore:
```diff
@@ -48,6 +48,7 @@ doxygen/
 .browse.VC.db*
 *.stackdump
 util/Win_Check_Output.txt
+*.vscode
```
Could you revert this?

OlliGranlund replied:
Will do

drashna commented on keyboards/crkbd/keymaps/olligranlund_nordic/keymap.c:
```diff
@@ -0,0 +1,131 @@
+#define _QWERTY 0
+#define _LOWER 1
+#define _RAISE 2
+#define _ADJUST 16
```
Just a heads up.... it's come up recently that having this set to 16 can increase your firmware size, just by having this at a high number. 

Ideally, this should be lower, especially if you're having issues with compiled firmware size. 

If that's not an issue here, then you can ignore this.

OlliGranlund replied:
It hasn't been an issue, but apparently something really got messed up since this keymap shoulda been under "iris" and now under "crkbd".... I really need to re-check this

drashna commented on keyboards/crkbd/keymaps/olligranlund_nordic/keymap.c:
```diff
@@ -0,0 +1,131 @@
+	)
+};
+
+void persistent_default_layer_set(uint16_t default_layer) {
```
Could you remove this function?  

It's been depreciated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

drashna commented on keyboards/dz60/dz60.h:
```diff
@@ -384,6 +384,33 @@
+ * │40  │41  │43  │46                      │4a  │4b  │4d  │4e  │
+ * └────┴────┴────┴────────────────────────┴────┴────┴────┴────┘
+*/
+#define LAYOUT_60_olligranlund_ISO( \
```
Is there a specific reason to add this over using the "all" layout? 

Also, we use snake case for the layout macros:
```suggestion
#define LAYOUT_60_olligranlund_iso( \
```

OlliGranlund replied:
Not sure how the ISO-enter is wired in the PCB, so it wasn't clear how you could achieve ISO + split backspace + split right shift. The layout I created solves this issue.

drashna commented on keyboards/satan/keymaps/olligranlund_iso/keymap.c:
```diff
@@ -48,6 +48,10 @@
 #define MYCM KC_MYCM  // Open default file manager
 
+// increase readability
+#define _______  KC_TRNS
```
These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

drashna commented on keyboards/satan/keymaps/olligranlund_iso/rules.mk:
```diff
@@ -13,5 +13,5 @@ MIDI_ENABLE = no            # MIDI controls
 UNICODE_ENABLE = no         # Unicode
 BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
-RGBLIGHT_ENABLE = yes        # Enable WS2812 RGB underlight. 
+RGBLIGHT_ENABLE = yes        # Enable WS2812 RGB underlight.  Do not enable this with audio at the same time.
```
This warning is incorrect, and has not been valid for a long time. 

```suggestion
RGBLIGHT_ENABLE = yes        # Enable WS2812 RGB underlight. 
```

Eg, you can absolutely enable both functions at the same time.

drashna commented:
You've opened this pull request from your fork's `master` branch. While this isn't a major issue now, later changes may result in merge conflicts if you try to file another PR because the commit history will have diverged between your master and QMK's. It is highly recommended for QMK development - regardless of what is being done or where - to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your master updated here:

[**Best Practices: Your fork's master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_best_practices?id=your-fork39s-master-update-often-commit-never)

If you need any help with this just ask.

You do not need to close this PR. Further changes should still be pushed to your master branch, until this PR is merged.

OlliGranlund commented:
Alright, so apparently my iris keymap had gotten in as a duplicate to the wrong folder, no idea how this happened... I removed that since it was irrelewant for crkbd. Reviewed the changes you requested and made changes accordingly. I had to create a new DZ60 layout due to there not being a ISO layout with split backspace and split right shift. I now added also the json-data of my layout.  Also cleaned layout to match 1:1 the visual representation of layouts.

I couldn't remember how I did the PR last time so that's why some of this mess. Should I now wait for you to accept this PR or should I create a new branch and push it to qmk-repo and request that to be merged?

mechmerlin commented:
Unless this new layout name is a community accepted standard, I am opposed to having this new layout named with your user name. Can you think of a more generic name for this or make any of the existing layout macros work with your keymap?

OlliGranlund commented:
I could come up with a better name for it, however I've just followed how the previous layouts have been made eg. "LAYOUT_60_tsangan_hhkb" has exactly as much information as my layout. Do you still feel like I should rename my layout to something like "LAYOUT_60_iso_split_backspace_rshift"?

noroadsleft commented:
> LAYOUT_60_iso_split_backspace_rshift

Most other 60% keyboards that I know of would use `LAYOUT_60_iso_split_bs_rshift` (`bs` being Backspace here) for that. You've got split Space as well, but I don't know that there's a custom for that as far as naming.

OlliGranlund commented:
Alrighty! Renamed the layout to something more generic, so I guess this should be fine now? How should we proceed with this merge, it's fine like this or should I do these changes via a separate branch and make PR for that?

drashna commented:
Could you remove the `.vscode/ipch/50bbb6394ee57e02/DZ60.ipch` file?

OlliGranlund commented:
Damn, did that get in there now too... Well I removed the file as requested.

zvecr commented on keyboards/satan/keymaps/olligranlund_iso/keymap.c:
```diff
@@ -48,6 +48,9 @@
 #define MYCM KC_MYCM  // Open default file manager
 
+// increase readability
+#define XXXXX    KC_NO
```
Is this required when `XXXXXXX` already exists?

```suggestion
```

drashna replied:
Well, it's not predefined, like `XXXXXXX`, so it depends on what you want.

Ideally, I think we should use the built in one.

noroadsleft replied:
I would also prefer use of the seven-character alias defined in QMK core.

noroadsleft replied:
@OlliGranlund I recommend you rewrite this keymap to use the `LAYOUT_60_iso` macro. (Your existing readme suggests that's the layout you're running anyway.) Right now this keymap is based on outdated conventions, and wouldn't compile from a current version of QMK if merged as-is.

drashna commented on keyboards/dz60/dz60.h:
```diff
@@ -384,6 +384,33 @@
     { k20,   KC_NO, k22,   k23,  k24,   k25,   k26,  k27,   k28,   k29,   k2a,   k2b,  k2c,   k2d,  KC_NO }, \
     { k30,   KC_NO, k32,   k33,  k34,   k35,   k36,  k37,   k38,   k39,   k3a,   k3b,  KC_NO, k3d,  k3e   }, \
     { k40,   k41,   KC_NO, k43,  KC_NO, KC_NO, k46,  KC_NO, KC_NO, KC_NO, KC_NO, k4b,  KC_NO, k4d,  k4e   }  \
+}^
```
```suggestion
}
```

OlliGranlund commented:
As my GH60 if fried, I can't test the compiled hex if it works as expected...

drashna commented:
Oh no, I'm very sorry to hear that! :(

OlliGranlund commented:
It's been a while. Yeah, my GH60 board is not working anymore and nothing seems to help for it, hence I would rather not do any more changes to my Satan keymap as it might break things. I've done the changes as requested, how should we go on with this merge? Do you want me to make it into a separate branch and PR or is it fine just in here after all the changes?

drashna commented:
This should be fine.  @mechmerlin @noroadsleft any issues with the layout changes?

OlliGranlund commented:
This still seems to be pending :)

mechmerlin commented:
> This should be fine. @mechmerlin @noroadsleft any issues with the layout changes?

looks good to me!

noroadsleft commented:
> This should be fine. @mechmerlin @noroadsleft any issues with the layout changes?

The DZ60 layout changes are good, but the Satan keymap actually won't compile as-is when this is merged into master.

noroadsleft commented:
> It's been a while. Yeah, my GH60 board is not working anymore and nothing seems to help for it, hence I would rather not do any more changes to my Satan keymap as it might break things. I've done the changes as requested, how should we go on with this merge? Do you want me to make it into a separate branch and PR or is it fine just in here after all the changes?

I forgot that your Satan is out of commission. Please disregard my comments regarding rewriting that keymap.

OlliGranlund commented:
Hmm, so we are still waiting for yanfali:s approval?

drashna commented:
Thanks!
